### PR TITLE
#422 delete wp modal

### LINKED ISSUE #422: Admin Tools - Create Delete Work Package Modal
### Description

Blocked by #420

Create a modal that is triggered by pressing delete on a selected work package page.

### Acceptance Criteria

Work Package Deletion Popup

- Open a modal to confirm deletion
- Top line of the modal should say; "Do you want to delete the work package <work_package_name> of project <project_name>?"
- Below this, let the user know; "This work package will be deleted."
- Below that; let the user know that "This action cannot be undone!"
- Block the user from being able to click confirm until they type in yes, delete, or confirm to confirm the deletion
- At the bottom of the modal; give the option to delete, or cancel the deletion.

### Proposed Solution

- Information listed above
- Can be based off other Modals like the add proposed solution modal.

### Mocks

image

image

image

### PR DESCRIPTION
**Pull request by nezamj03**
## Changes

In a WP page, the delete button is added into the dropdown, where a modal will prompt confirmation of deletion.

## Screenshots
Screen Shot 2023-02-27 at 2 34 50 AM
Screen Shot 2023-02-27 at 2 35 20 AM
Screen Shot 2023-02-27 at 2 35 38 AM
Screen Shot 2023-02-27 at 2 35 49 AM
Screen Shot 2023-02-27 at 2 36 11 AM
Screen Shot 2023-02-27 at 2 37 07 AM
Screen Shot 2023-02-27 at 2 38 28 AM

## To Do

_Any remaining things that need to get done_

- testing

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [x] All commits are tagged with the ticket number
- [x] No linting errors / newline at end of file warnings
- [x] All code follows repository-configured prettier formatting
- [x] No merge conflicts
- [x] All checks passing
- [x] Screenshots of UI changes (see Screenshots section)
- [x] Remove any non-applicable sections of this template
- [x] Assign the PR to yourself
- [x] No `yarn.lock` changes (unless dependencies have changed)
- [x] Request reviewers & ping on Slack
- [x] PR is linked to the ticket (fill in the closes line below)

Closes #422

### CODE REVIEW

File: src/frontend/src/pages/WorkPackageDetailPage/DeleteWorkPackageModalContainer/DeleteWorkPackage.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+}: DeleteWorkPackageRequestProps) => {
+  const history = useHistory();
+  const toast = useToast();
+  const { isLoading, isError, error, mutateAsync } = useDeleteWorkPackage();
+
+  const handleConfirm = async ({ wbsNum }: DeleteWorkPackageInputs) => {
+    handleClose();
+    const wbsNumber = validateWBS(wbsNum);
+    await mutateAsync(wbsNumber).catch((error) => {
+      if (error instanceof Error) {
```
Comment (Peyton-McKee) [inline src/frontend/src/pages/WorkPackageDetailPage/DeleteWorkPackageModalContainer/DeleteWorkPackage.tsx:37]: I dont think u need this cast

File: src/frontend/src/pages/WorkPackageDetailPage/WorkPackagePage.tsx [hunk 2]
```diff
@@ -23,6 +23,7 @@ const WorkPackagePage: React.FC<WorkPackagePageProps> = ({ wbsNum }) => {
   const [editMode, setEditMode] = useState<boolean>(query.get('edit') === 'true');
   const auth = useAuth();
   const isGuest = auth.user?.role === 'GUEST';
+  const isAdmin = auth.user?.role === 'ADMIN' || auth.user?.role === 'APP_ADMIN';
```
Comment (Peyton-McKee) [inline src/frontend/src/pages/WorkPackageDetailPage/WorkPackagePage.tsx:26]: Probably should validate auth.user in the isloading if statement then move these two things after

**Review by Peyton-McKee — CHANGES_REQUESTED** [review]
Looks great!

File: src/frontend/src/pages/WorkPackageDetailPage/DeleteWorkPackageModalContainer/DeleteWorkPackage.tsx (hunk 1 above)
Comment (anthonybernardi) [reply src/frontend/src/pages/WorkPackageDetailPage/DeleteWorkPackageModalContainer/DeleteWorkPackage.tsx:37]: you do

File: src/frontend/src/pages/WorkPackageDetailPage/DeleteWorkPackageModalContainer/DeleteWorkPackageView.tsx [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+interface DeleteWorkPackageViewProps {
+  workPackage: WbsNumber;
+  modalShow: boolean;
+  onHide: () => void;
+  onSubmit: (data: DeleteWorkPackageInputs) => Promise<void>;
+}
+
+const DeleteWorkPackageView: React.FC<DeleteWorkPackageViewProps> = ({ workPackage, modalShow, onHide, onSubmit }) => {
+  const workPackageWbsTester = (wbsNum: string | undefined) => wbsNum !== undefined && wbsNum === wbsPipe(workPackage);
+  // const workPackageWbsTester = (wbsNum: string | undefined) => true;
```
Comment (anthonybernardi) [inline src/frontend/src/pages/WorkPackageDetailPage/DeleteWorkPackageModalContainer/DeleteWorkPackageView.tsx:35]: delete

File: src/frontend/src/pages/WorkPackageDetailPage/WorkPackageViewContainer/WorkPackageViewContainer.tsx [hunk 4]
```diff
@@ -100,6 +110,14 @@ const WorkPackageViewContainer: React.FC<WorkPackageViewContainerProps> = ({
       Stage Gate
     </MenuItem>
   );
+  const deleteBtn = (
```
Comment (anthonybernardi) [inline src/frontend/src/pages/WorkPackageDetailPage/WorkPackageViewContainer/WorkPackageViewContainer.tsx:113]: deleteButton

File: src/frontend/src/pages/WorkPackageDetailPage/WorkPackageViewContainer/WorkPackageViewContainer.tsx [hunk 5]
```diff
@@ -76,6 +81,11 @@ const WorkPackageViewContainer: React.FC<WorkPackageViewContainerProps> = ({
     handleDropdownClose();
   };
 
+  const handleClickDelete = () => {
+    setShowDeleteModal(true);
+    handleDropdownClose();
+  };
+
   const editBtn = (
```
Comment (anthonybernardi) [inline src/frontend/src/pages/WorkPackageDetailPage/WorkPackageViewContainer/WorkPackageViewContainer.tsx:89]: can u change this to editButton pls

File: src/frontend/src/pages/WorkPackageDetailPage/WorkPackageViewContainer/WorkPackageViewContainer.tsx [hunk 6]
```diff
@@ -128,6 +146,7 @@ const WorkPackageViewContainer: React.FC<WorkPackageViewContainerProps> = ({
         {workPackage.status === WbsElementStatus.Inactive ? activateBtn : ''}
         {workPackage.status === WbsElementStatus.Active ? stageGateBtn : ''}
         {createCRBtn}
```
Comment (anthonybernardi) [inline src/frontend/src/pages/WorkPackageDetailPage/WorkPackageViewContainer/WorkPackageViewContainer.tsx:148]: and this to createChangeRequestButton

**Review by anthonybernardi — CHANGES_REQUESTED** [review]
awesome! looks great to me

File: src/frontend/src/pages/WorkPackageDetailPage/DeleteWorkPackageModalContainer/DeleteWorkPackage.tsx [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+  handleClose
+}: DeleteWorkPackageRequestProps) => {
+  const history = useHistory();
+  const toast = useToast();
+  const { isLoading, isError, error, mutateAsync } = useDeleteWorkPackage();
+
+  const handleConfirm = async ({ wbsNum }: DeleteWorkPackageInputs) => {
+    handleClose();
+    const wbsNumber = validateWBS(wbsNum);
+    await mutateAsync(wbsNumber).catch((error) => {
```
Comment (anthonybernardi) [inline src/frontend/src/pages/WorkPackageDetailPage/DeleteWorkPackageModalContainer/DeleteWorkPackage.tsx:36]: this won't work, do something like this:
```
    try {
      const wbsNumber = validateWBS(wbsNum);
      await mutateAsync(wbsNumber);
      handleClose();
      history.goBack();
      toast.success(`Project #${wbsPipe(wbsNumber)} Deleted Successfully!`);
    } catch (e) {
      if (error instanceof Error) {
        toast.error(error.message);
      }
    }
```

**Review by anthonybernardi — CHANGES_REQUESTED** [review]
nice

**Review by anthonybernardi — APPROVED** [review]

**Pull request merged**